### PR TITLE
Clean identity server

### PR DESCRIPTION
**Pull request by h1ppox99**
Conclusion of the cleaning :
1 - it is necessary to tackle #61 - key management is not good for the moment 
2 - maybe a deep review of cron tasks will be needed (not enough test coverage for the moment)

### CODE REVIEW

File: packages/matrix-identity-server/src/config.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "database_ssl": false,
   "database_user": "",
   "database_vacuum_delay": 3600,
-  "federated_identity_services": null,
+  "federated_identity_services": ["-invalid.host.com"],
```
Comment (guimard) [inline packages/matrix-identity-server/src/config.json:13]: Why this change ? null means no fd service

File: packages/matrix-identity-server/src/db/sql/sqlite.ts [hunk 2]
```diff
@@ -189,10 +191,7 @@ class SQLite<T extends string> extends SQL<T> implements IdDbBackend<T> {
       )
 
       stmt.finalize((err) => {
-        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-        if (err) {
-          reject(err)
-        }
+        reject(err)
```
Comment (guimard) [inline packages/matrix-identity-server/src/db/sql/sqlite.ts:194]: More short version :  `stmt.finalize(reject)`

File: packages/matrix-identity-server/src/config.json (hunk 1 above)
Comment (h1ppox99) [reply packages/matrix-identity-server/src/config.json:13]: i made this change to have a better test coverage - invalid host names are filtered out from the config when creating the MatrixIdentityServer

Comment (guimard) [reply packages/matrix-identity-server/src/config.json:13]: Not the good file. Here is the default config
